aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SdkConfigDecorator.kt

```diff
@@ -59,6 +61,31 @@ object SdkConfigCustomization {
             "map" to mapBlock,
         )
     }
+
+    fun copyFieldAndCheckForServiceConfig(
+        fieldName: String,
+        map: Writable?,
+    ) = adhocCustomization<SdkConfigSection.CopySdkConfigToClientConfig> { section ->
+        val mapBlock = map?.let { writable { rust(".map(#W)", it) } } ?: writable { }
+        val envKey = "AWS_${fieldName.toSnakeCase().uppercase()}".dq()
+        val profileKey = fieldName.toSnakeCase().dq()
+
+        rustTemplate(
+            """
+            if ${section.sdkConfig}.get_origin("endpoint_url").is_client_config() {
+                ${section.serviceConfigBuilder}.set_$fieldName(${section.sdkConfig}.$fieldName()#{map});
+            } else {
+                ${section.serviceConfigBuilder}.set_$fieldName(
+                    ${section.sdkConfig}
+                        .service_config()
+                        .and_then(|conf| conf.load_config(service_config_key($envKey, $profileKey)).map(|it| it.parse().unwrap()))
+                        .or_else(|| ${section.sdkConfig}.$fieldName()#{map})
```

Comment:
If I understand it correctly, `shared_environment_variable` (a.k.a global env) takes precedence over `service_profile_file`. If a user configures a config value in those two places only, will this code respect `shared_environment_variable`?

Reply:
yeah

Reply:
That right, I misread the code. `${section.sdkConfig}.service_config()` returns `EnvServiceConfig` (not service-specific-only config), so it knows how to handle the said precedence.